tasks/ThrusterTask.cpp

```diff
@@ -95,14 +93,13 @@ void ThrusterTask::cleanupHook()
     node->Fini();
 }
 
-void ThrusterTask::setGazeboModel(WorldPtr _world,  ModelPtr _model, sdf::ElementPtr _plugin)
+void ThrusterTask::setGazeboThruster( ModelPtr model )
```

Comment:
I've changed that because there was no need to receive **_world** and **_plugin**
Now, each task has a different interface: 

    setGazeboWorld(physics::WorldPtr _world)
    setGazeboModel(WorldPtr _world,  ModelPtr _model)
    setGazeboThruster( ModelPtr model )
    setGazeboLaserScan( ModelPtr model, string sensorName )

What should I do ?
What do you mean with **true plugin-based system** ?